webapp/utils/aws/cloudwatch.py

```diff
@@ -39,9 +39,21 @@ def client(self) -> "CloudWatchLogsClientType":
     def get_log_messages(self, task_id: str) -> list:
         messages: list = []
         if logs := self.get_log_events(task_id):
-            messages.extend(event["message"] for event in logs)
+            return self.get_log_summary(logs)
         return messages
 
+    def get_log_summary(self, logs: list[dict]) -> list[str]:
```

Comment:
I probably sound like a broken record, but I think a little docstring could be helpful here.

This is a super elegant update (nice work!), but I'll admit it took me a bit to parse the function and understand how it worked (that given a target log line, we would _then_ return all lines after that via `logs[summer_index:]]`).

But again, seems like a great solution, and improved run details!  Just think a little blurb here explaining could be helpful.

Reply:
Agreed!